pkg/components/ldap_auth.go

```diff
@@ -0,0 +1,60 @@
+package components
+
+import (
+	"context"
+	"net/http"
+)
+
+type authHeader struct {
+	Wrapped http.RoundTripper
+}
+
+// temp function for searching through a slice until we move the whitelist into a map
+func Contains(s []string, target string) bool{
+	for _, c := range s{
+		if target == c {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *authHeader) RoundTrip(req *http.Request) (*http.Response, error){
+	incomingLdapGroups := req.Header["X-Slauth-User-Groups"]
```

Comment:
SLAuth is a tool that is used internally only. Since transportd is one of our open sourced projects, I see a couple of options, though there are probably more.

1) make this a generic header validation component which expects certain header key/value pairs. Our services can use this to then add checks for the specific slauth headers

2) make an internal extention of transportd, similar to how serverfull-gateway extends transportd. However, this internal extension would have Atlassian specific components.

What do you think?

Reply:
Personally, I prefer the first option of making the header name configurable as well as the allowed values.

Reply:
I'm shooting for the first option, I think I can include another param to make this header name configurable

Reply:
done